specs/hierarchical_consensus.md

```diff
@@ -0,0 +1,1079 @@
+- __Separator (1 byte)__: One character used to separate the Subnet ID from the raw Filecoin address. The specific separator used for HC is the`:` string.
+- __RawAddress (up to 66 bytes)__: Byte representation of the raw Filecoin address. The maximum size is determined by the size of the SECPK address.
+
+> Structure of Hierarchial Address
+```
+|----------|------------------------------------------------------------------------------------------------|
+| protocol |        payload                                                                                 |
+|----------|------------------------------------------------------------------------------------------------|
+|    4     | subnetID (up to 74 bytes) | separator (1 byte) | raw address (up to 66 bytes) |
+```
+
+The payload of an HA looks something like:
+```
+| /root/f01001 | : | <payload raw address bytes> |``
+```
+
+With hierarchical addresses two new functions are introduced to Filecoin addresses:
+- `Raw()`: Returns the raw address of the hierarchical address. If it is not an HA, it still returns the corresponding raw address.
+- `Subnet()`: It returns the subnet ID of the HA. It returns an error or an `Subnet.Undef` if the address is not an HA.
+- `PrettyString()`: Returns a beutified version of an HA address like `/root/f0100:<raw address>`. This is method is only available for HA. For the rest of addresses it just returns `String()`. 
+
+Finally, a pair of keys from a user control the same address in every subnet in the system. Thus, the raw address determines the ownership of a specific HA. 
+
+## Checkpointing
+Checkpoints are used to anchor a subnet's security to that of its parent network, as well as to propagate information from a child chain to other subnets in the system. Checkpoints for a subnet can be verified at any point using the state of the subnet chain which can then be used to generate proofs of misbehaviors in the subnet (or so-called _fraud/fault proofs_) which, in turn, can be used for penalizing misbehaving entities (_"slashing"_). See [detectable misbehaviors](#Detectable-misbehaviors "Detectable misbehaviors") for further details.
+
+Checkpoints need to be signed by miners of a child chain and committed to the parent chain through their corresponding `SA`. The specific signature policy is defined in the `SA` and determines the type and minimum number of signatures required for a checkpoint to be accepted and validated by the `SA` for its propagation to the top chain. Different signature schemes may be used here, including multi-signatures or threshold signatures among subnet miners. For instance, in the reference implementation of `SA`, the actor waits for `2/3` of the validators in the subnet to send the valid checkpoint signed before propagating it to the `SCA` for commitment. Alternative verification policies for a subnet can be implemented in the `SubmitCheckpoint()` function of the subnet actor interface. 
```

Comment:
menion and link to https://github.com/filecoin-project/mir for "reference implementation of SA" also it should be more than `2/3` and not `2/3`

Reply:
A concise notation I have seen for that is `+2/3`.